File: src/Entity/Maintenance.php
<?php

namespace App\Entity;
use App\Repository\MaintenanceRepository;
use App\Entity\Vehicule;
use Symfony\Component\Validator\Constraints as Assert;


use Doctrine\ORM\Mapping as ORM;


 

 #[ORM\Entity(repositoryClass: MaintenanceRepository::class)]
 
class Maintenance
{
   

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idMaintenance = null ;
   
    

    #[ORM\Column(length:50)]
    #[Assert\NotBlank(message: "Remplir vos champs")]
    #[Assert\Length(max: 10)]
    private ?string $etat = null ;
    

    #[ORM\Column(length:50)]
    #[Assert\NotBlank(message: "Remplir vos champs")]
    #[Assert\Length(min: 5)]
    private ?string $nomSosRep = null ;
    
   
    #[ORM\ManyToOne(targetEntity: Vehicule::class)]
    #[ORM\JoinColumn(name: "id_vehicule", referencedColumnName: "id_vehicule")]
    protected $idVehicule;
    



    
    public function getIdMaintenance(): ?int
    {
        return $this->idMaintenance;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getNomSosRep(): ?string
    {
        return $this->nomSosRep;
    }

    public function setNomSosRep(string $nomSosRep): self
    {
        $this->nomSosRep = $nomSosRep;

        return $this;
    }

    public function getIdVehicule(): ?Vehicule
    {
        return $this->idVehicule;
    }

    public function setIdVehicule(?Vehicule $idVehicule): self
    {
        $this->idVehicule = $idVehicule;

        return $this;
    }


}

File: src/com/mycompany/myapp/entities/Maintenance.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author hp
 */
public class Maintenance {
    
    int id_maintenance ;
    String nom_sos_rep , etat ;
    int id_vehicule ; 

    public Maintenance(int id_maintenance, String nom_sos_rep, String etat, int id_vehicule) {
        this.id_maintenance = id_maintenance;
        this.nom_sos_rep = nom_sos_rep;
        this.etat = etat;
        this.id_vehicule = id_vehicule;
    }

    public Maintenance() {
    }

    public Maintenance(String nom_sos_rep, String etat, int id_vehicule) {
        this.nom_sos_rep = nom_sos_rep;
        this.etat = etat;
        this.id_vehicule = id_vehicule;
    }

    public int getId_maintenance() {
        return id_maintenance;
    }

    public String getNom_sos_rep() {
        return nom_sos_rep;
    }

    public String getEtat() {
        return etat;
    }

    public int getId_vehicule() {
        return id_vehicule;
    }

    public void setId_maintenance(int id_maintenance) {
        this.id_maintenance = id_maintenance;
    }

    public void setNom_sos_rep(String nom_sos_rep) {
        this.nom_sos_rep = nom_sos_rep;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public void setId_vehicule(int id_vehicule) {
        this.id_vehicule = id_vehicule;
    }

    @Override
    public String toString() {
        return "Maintenance{" + "id_maintenance=" + id_maintenance + ", nom_sos_rep=" + nom_sos_rep + ", etat=" + etat + ", id_vehicule=" + id_vehicule + '}';
    }
    
    
    

   
}
